src/core/public/chrome/README.md

```diff
@@ -0,0 +1,275 @@
+- NavLinksService : for manipulating nav links.
+- RecentlyAccessedService : for recently accessed history.
+- DocTitleService: for accessing and updating the document title
+- UI : All the UI components,icons for e.g  header, loaders.
+```
+### NavControlsService : 
+- Interface : ChromeNavControls
+- Signature : ```navControls: ChromeNavControls```
+- Registering new controller 
+
+Example : 
+Register a left-side nav control rendered with React.
+
+```jsx
+chrome.navControls.registerLeft({
+  mount(targetDomElement) {
+    ReactDOM.mount(<MyControl />, targetDomElement);
+    return () => ReactDOM.unmountComponentAtNode(targetDomElement);
+  }
+})
+```
+### NavLinksService : 
+- Interface :  ChromeNavLinks
+- Signature : ```navLinks: ChromeNavLinks```
+- e.g Method : 
+
+```typescript
+Get an observable for a sorted list of navlinks :-
+getNavLinks$(): Observable<Array<Readonly<ChromeNavLink>>>;
+
+Get the state of a navlink at this point in time :-
+get(id: string): ChromeNavLink | undefined;
+
+Get the current state of all navlinks :-
+getAll(): Array<Readonly<ChromeNavLink>>;
+
+Check whether or not a navlink exists :-
+has(id: string): boolean;
+
+Remove all navlinks except the one matching the given id :-
+showOnly(id: string): void;
+```
+- How to access
+  ###### Get the current state of all navlinks: 
+  ```ts
+  core.chrome.navLinks.getAll() 
+  ```
+
+  ###### Get the state of a navlink at this point in time: 
+  ```ts
+  core.chrome.navLinks.get()
+  ```
+
+### RecentlyAccessedService : 
+
+- Interface : ChromeRecentlyAccessed
+- Signature : ```recentlyAccessed: ChromeRecentlyAccessed```
+- The Recently viewed items are stored in the browser's local storage.
+- You can go back to the recent search/visualization/dashboard , each item has (link,label,id)
+- Methods :
+
+```typescript
+
+Adds a new item to the recently accessed history :-
+add(link: string, label: string, id: string): void;
+ 
+Gets an Array of the current recently accessed history :-
+get(): ChromeRecentlyAccessedHistoryItem[];
+
+Gets an Observable of the array of recently accessed history :-
+get$(): Observable<ChromeRecentlyAccessedHistoryItem[]>; 
+```
+- How to access
+   ###### Adds a new item to the recently accessed history :
+   ```ts
+   chrome.recentlyAccessed.add('/app/map/1234', 'Map 1234', '1234');
+   ```
+   ###### Gets an Array of the current recently accessed history :
+   ```ts
+    chrome.recentlyAccessed.get().forEach(console.log);;
+   ```
+   
+### DocTitleService : 
+- Interface : ChromeDocTitle
+- Signature : ```docTitle: ChromeDocTitle```
+  ###### - How to change the title of the document
+
+
+   ```ts
+   chrome.docTitle.change('My application title')
+   chrome.docTitle.change(['My application', 'My section'])
+   ```
+### UI :
+###### consists of tsx/scss files && renders UI components from css Library e.g ```<Progress props={props}>```
+
+- Adding/overriding existing css : e.g.
+  ```jsx
```

Comment:
Can we limit the code snippet markdown to just code examples? Its a lil unreadable when the snippet markdown is used for general text :(

Reply:
Makes sense!